runners/flink/runner/src/main/java/org/apache/beam/runners/flink/FlinkStreamingTransformTranslators.java

```diff
@@ -881,7 +1005,81 @@ public void flatMap(
                   in.getValue().getKey(),
                   in.withValue(in.getValue().getValue()));
 
-          in.withValue(workItem);
+          out.collect(in.withValue(workItem));
```

Comment:
It's not really a bug since it doesn't affect correctness but simply does an extra copy.